### DISCUSSION TITLE
Snipe-IT Email configuration error - Mail could not be sent. No additional error message provided. Check your mail settings and your app log.

### DISCUSSION THREAD

**Discussion by thilinadias** [snipe/snipe-it#14041] [Q&A]
✏️ Hi Github peeps, Welcome to my discussion regarding Snipe-IT Email configuration error - **Mail could not be sent. No additional error message provided. Check your mail settings and your app log.**

If you come across mentioned issue when setting up the Snipe-IT, Do the following steps. 

1. Check your email configurations are correct with the email service that you use, on **.env** file in your **"snipe-it"** directory. [Normally on **`cd /var/www/html/snipe-it`** (It might be different in your environment]
2. Check the firewall settings on your web server [**Ubuntu - '`sudo ufw status`'**] This might be interfering with emails
3. Telnet to manually test the email configuration. [**telnet your-smtp-server.com your-port**] ex: `telnet smtp.gmail.com 587`. If **it is successful** you might see the following message. "220 smtp.gmail.com ESMTP [...]"
4. Then check the application logs on Snipe-IT, whether it has recorded a possible error that was occurred. Normally on following location, 

**`cd /var/www/html/snipe-it/storage/logs`

tail -f laravel.log** You might see following error. **"production.ERROR: Connection could not be established with host smtp-mail.outlook.com :stream_socket_client(): unable to connect to starttls://smtp-mail.outlook.com:587 (Unable to find the socket transport &quot;starttls&quot; - did you forget to enable it when you configured PHP?) {"userId":1,"exception":"[object] (Swift_TransportException(code: 0): Connection could not be established with host smtp-mail.outlook.com :stream_socket_client(): unable to connect to starttls://smtp-mail.outlook.com:587 (Unable to find the socket transport &quot;starttls&quot; - did you forget to enable it when you configured PHP?) at /var/www/snipe-it/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/StreamBuffer.php:261)
[stacktrace]**

5. It means that, in the **php.in**i file you doesn't enable the **tls** service. The location of this file may vary depending on your PHP installation, but common locations are **`/etc/php/7.x/cli/php.ini`** and **`/etc/php/7.x/apache2/php.ini`.**

6. Look for the extension= or ;extension= lines related to OpenSSL. Uncomment (remove the ; at the beginning) or add the following lines:

```
extension=openssl
extension=sockets
```

7. Save the changes and restart your web server:

`sudo service apache2 restart`   For Apache

`sudo service nginx restart`     For Nginx

8. After making these changes, run a PHP script to check if STARTTLS is now supported. Create a file (e.g., check_starttls.php) with the following content:

```
<?php
$transports = stream_get_transports();

foreach ($transports as $transport) {
    echo $transport . PHP_EOL;
}
```

9. Save the changes and run the script

`php check_starttls.php`

10. Then you can see the supported transport including TLS. 

11. After that restart the apache server. 

`sudo service apache2 restart`  # For Apache
# or
`sudo service nginx restart`    # For Nginx

12. Again try to send the email and it will be successful.

Thank you guys! Good Luck! 😉🔎

**Comment by snipe** [comment]
Hi there - can you show me your mail section of your `.env` (with passwords redacted, of course)

**Comment by cevallose** [comment]
I got the same problem here . Some one help me please

**Reply by simisola88** [reply]
@cevallose  as mentioned by @thilinadias 
confirm you can reach the  smtp host and port.. also confirm your  smtp username and  password are good

**Comment by MaKaNu** [comment]
Struggle with simmilar issue on a docker instance aswell on our new testing instance on kubernetes. With debug enabled I could track it down to:

```
Swift_TransportException: Connection could not be established with host 10.3.0.8 :stream_socket_client(): php_network_getaddresses: getaddrinfo for &lt;nil&gt; failed: Name or service not known in /var/www/html/vendor/swiftmailer/swiftmailer/lib/classes/Swift/Transport/StreamBuffer.php
```

The SMTP Host is a simple postfix mail relay so the config looks like this in the helm chart:

```yaml
MAIL_DRIVER: smtp
MAIL_HOST: 10.3.0.8
MAIL_PORT: 25
MAIL_ENCRYPTION: null
```

**Reply by snipe** [reply]
Seems there's an issue with the DNS on your server, meaning it's unable to translate the hostname 10.3.0.8 into a valid IP address. You'll want to check that the IP address is valid and that there are no firewall issues.

**Reply by MaKaNu** [reply]
Well 10.3.0.8 is the ip address and checking from inside the kubenetes container the name resolution with nslookup works results in 10.3.0.8 and also calling `curl -v 10.3.0.8:25` connects successful.

I think the weird part is the `getaddrinfo for &lt;nil&gt;`

**Reply by MaKaNu** [reply]
I might have found the issue, but still wonder why. On kubernetes I launched with 6.3.3 because this was the version we migrated from the last time with our production (had a few hardware losses, and dont want to fight on two battlegrounds at the same time.) It seems to be the issue that the mail.php in v6.3.3 created the problem. I am not sure if it is worth to patch 6.3.3 (or if it is already patched.) I now tried v7.1.16 (v8.1.15 didn't launched different topic) and it seems to work.

**Reply by snipe** [reply]
@uberbrady any thoughts on this?
